pkg/apis/application/v1alpha1/types.go

```diff
@@ -3626,17 +3626,11 @@ func (c *Cluster) RawRestConfig() (*rest.Config, error) {
 
 	switch {
 	case c.Server == KubernetesInternalAPIServerAddr && env.ParseBoolFromEnv(EnvVarFakeInClusterConfig, false):
-		conf, exists := os.LookupEnv("KUBECONFIG")
-		if exists {
-			config, err = clientcmd.BuildConfigFromFlags("", conf)
-		} else {
-			var homeDir string
-			homeDir, err = os.UserHomeDir()
-			if err != nil {
-				homeDir = ""
-			}
-			config, err = clientcmd.BuildConfigFromFlags("", filepath.Join(homeDir, ".kube", "config"))
-		}
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		config, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			loadingRules,
+			&clientcmd.ConfigOverrides{},
+		).ClientConfig()
```

Comment:
how are we fetching the kubeconfig file path here? From the code you've removed, i can see that we aren't utilising the env variable anymore.